### ISSUE TITLE
Mock a trait with a constructor

### ISSUE THREAD

**Issue by konstantin-mueller** [ryo33/mry#16]
How can I mock a trait with a constructor?

I have this example trait:
```
#[cfg_attr(test, mry::mry)]
trait MyTrait {
    fn new() -> Self;
}
```

And I'm getting this error:
```
error[E0424]: expected value, found module `self`
  --> src/trait.rs:35:18
   |
35 | #[cfg_attr(test, mry::mry)]
   |                  ^^^^^^^^ `self` value is a keyword only available in methods with a `self` parameter
36 | trait MyTrait {
37 |     fn new() -> Self;
   |        --- this function doesn't have a `self` parameter
   |
   = note: this error originates in the attribute macro `mry::mry` (in Nightly builds, run with -Z macro-backtrace for more info)
help: add a `self` receiver parameter to make the associated `fn` a method
   |
37 |     fn new&self() -> Self;
   |           +++++
```

**Comment by ryo33** [comment]
I've released a new version `mry@0.10.0` on crates.io with the support of static methods in trait. Could you try it?

**Comment by konstantin-mueller** [comment]
Thank you, it works. But now I have a problem with generics in a trait:

```
#[cfg_attr(test, mry::mry)]
trait Test {
    async fn test<T: Send>(&self, value: T);
}
```

The error is:

```
error[E0412]: cannot find type `T` in this scope
  --> src/trait.rs:37:42
   |
37 |     async fn test<T: Send>(&self, value: T);
   |                                                                   ^ not found in this scope
```
